mmdata.c

```diff
@@ -1101,6 +1101,39 @@ int nmbrEq(nmbrString *s,nmbrString *t)
 }
 
 
+/* Compute a hash of a string */
+/* This simply computes a XOR of the first numbers */
+int nmbrHash(nmbrString *s)
+{
+  static long salt[] = { 4938, 48977, 6897, 7293, 2663, 7925, 2999, 12238,
+      40033, 14038, 10699, 29746, 56108, 34526, 63576, 52053, 61949, 41177, 43740, 22822
+ };
+  long i;
+  long hash = 0;
+  i = -1;
+  while (i < 13 && s[i] != -1) {
+    hash ^= ( s[i] + salt[i] ) << i;
+    i++;
+  }
+  return hash;
+}
+
+
+/* Compare two sub-strings */
+/* Unlike strcmp, this returns a 1 if the strings are equal
+   and 0 otherwise. */
+/* Only the token is compared.  The whiteSpace string is
+   ignored. */
+int nmbrSubEq(nmbrString *s,long sstart, nmbrString *t, long tstart, long len)
+{
+  long i;
+  if (sstart - 1 + len > nmbrLen(s)
+   || tstart - 1 + len > nmbrLen(t)) return 0;
+  for (i = 0; s[sstart-1+i] == t[tstart-1+i] && i<len; i++);

```

Comment:
Suspicious index in first loop: -1, if sstart == 0.  (1) Out of bounds access to s and t, or (2) Parameters sstart and tstart must be > 0.

Either provide parameter checks (cf. line 1211), or (minimum) state limitations in comment line 1122